### PR TITLE
Fix for legacy named arguments

### PR DESCRIPTION
**Pull request by vladmos**
Replaces legacy keyword arguments with positional arguments in some builtin functions, e.g. `bool(x = foo)` -> `bool(foo)`.

bazelbuild/bazel#5010

### CODE REVIEW

File: warn/warn_bazel_api.go [hunk 1]
```diff
@@ -832,3 +832,105 @@ func ruleImplReturnWarning(f *build.File) []*LinterFinding {
...
+	"str":     "x",
+	"repr":    "x",
+	"bool":    "x",
+	"int":     "x",
+	"dir":     "x",
+	"type":    "x",
+	"hasattr": "x",
+	"getattr": "x",
+	"select":  "x",
+}
```
Comment (laurentlb) [inline warn/warn_bazel_api.go]: How are you going to fix functions with more than 1 argument?

Comment (vladmos) [reply warn/warn_bazel_api.go]: I was going to implement it in portions, but perhaps I should support flexible amount of arguments now and just add functions/methods later.

Comment (vladmos) [reply warn/warn_bazel_api.go]: Refactored to handle both legacy keyword and legacy positional parameters, even for the same function

File: warn/warn_bazel_api_test.go [hunk 2]
```diff
@@ -650,3 +650,67 @@ def macro():
...
+int(x = "13", base = 8)
+dir(x = foo)
+type(x = foo)
+hasattr(x = foo, name = "bar")
+getattr(x = foo, name = "bar", default = "baz")
+select(x = {})
+glob(["*.cc"], ["test*"])
+native.glob(["*.cc"], ["test*"])
+glob(["*.cc"])
+native.glob(["*.cc"])
```
[the remark this answers is not in the document]
Comment (vladmos) [reply warn/warn_bazel_api_test.go]: Done.

File: warn/warn_bazel_api_test.go [hunk 3]
```diff
@@ -650,3 +650,99 @@ def macro():
...
+list([1, 2, 3])
+len([1, 2, 3])
+str(foo)
+repr(foo)
+bool(3)
+int("3")
+int("13", base = 8)
+dir(foo)
+type(foo)
+hasattr(foo, name = "bar")
```
Comment (laurentlb) [inline warn/warn_bazel_api_test.go]: Should be: `hasattr(foo, "bar")`

File: warn/warn_bazel_api_test.go [hunk 4]
```diff
@@ -650,3 +650,99 @@ def macro():
...
+getattr(foo, name = "bar", default = "baz")
```
Comment (laurentlb) [inline warn/warn_bazel_api_test.go]: Should be:
`getattr(foo, "bar", "baz")`

File: warn/warn_bazel_api_test.go [hunk 5]
```diff
@@ -650,3 +650,99 @@ def macro():
...
+glob(include = [], exclude = [])
+glob([], exclude = [])
+glob([], [], 1)
+glob(*args, [])
+`, `
+glob(["*.cc"], exclude = ["test*"])
+glob(["*.cc"])
+glob([], exclude = [])
+glob([], exclude = [])
+glob([], [], 1)
```
Comment (laurentlb) [inline warn/warn_bazel_api_test.go]: Add exclude_directories:
`glob([], exclude = [], exclude_directories = 1)`

File: warn/warn_bazel_api_test.go (hunk 3 above)
Comment (vladmos) [reply warn/warn_bazel_api_test.go]: Done.

File: warn/warn_bazel_api_test.go (hunk 4 above)
Comment (vladmos) [reply warn/warn_bazel_api_test.go]: Done.

File: warn/warn_bazel_api_test.go (hunk 5 above)
Comment (vladmos) [reply warn/warn_bazel_api_test.go]: Done.

**Comment by vladmos** [comment]
I'll run it on the existing files right before importing.

**Pull request merged**